chain/client/src/view_client.rs

```diff
@@ -280,7 +280,7 @@ impl ViewClientActor {
             let cps: Vec<AccountId> = shard_ids
                 .iter()
                 .map(|&shard_id| {
-                    let cp = epoch_info.sample_chunk_producer(block_height, shard_id);
```

Comment:
Is it safe to `unwrap()` here? The function returns an Error, so maybe it would be better to return an error.
(Also applies to other `unwraps()` on the new `sample_chunk_producer`)